src/sync.c

```diff
@@ -95,9 +95,10 @@ static void adjust_ref(sync_t *st, unsigned int ref, int cfo)
     unsigned int n;
     float cfo_freq = 2 * M_PI * cfo * CP_FM / FFT_FM;
 
-    // sync bits (after DBPSK)
+    // differentially-encoded sync & parity bits
     static const signed char sync[] = {
-        -1, 1, -1, -1, -1, 1, 1
+        -1, 1, -1, -1, -1, 1, 1, 0, 1, -1, 0, 0, 0, -1, -1, 0,
+        0, 0, 0, 0, -1, 1, -1, 0, 0, 0, 0, 0, 0, 0, 0, -1
```

Comment:
Instead of checking only the first seven sync bits here, I added the remaining sync bits, as well as the parity bits (which are fixed after differential encoding).